pinot-query-planner/src/main/java/org/apache/pinot/query/parser/CalciteRexExpressionParser.java

```diff
@@ -99,31 +100,42 @@ private static List<Expression> convertDistinctSelectList(RexExpression.Function
     return selectExpr;
   }
 
-  public static List<Expression> convertOrderByList(List<RexExpression> rexInputRefs,
-      List<RelFieldCollation.Direction> directions, PinotQuery pinotQuery) {
-    List<Expression> orderByExpr = new ArrayList<>();
-
-    for (int i = 0; i < rexInputRefs.size(); i++) {
-      orderByExpr.add(convertOrderBy(rexInputRefs.get(i), directions.get(i), pinotQuery));
+  public static List<Expression> convertOrderByList(SortNode node, PinotQuery pinotQuery) {
+    List<RexExpression> collationKeys = node.getCollationKeys();
+    List<RelFieldCollation.Direction> collationDirections = node.getCollationDirections();
+    List<RelFieldCollation.NullDirection> collationNullDirections = node.getCollationNullDirections();
+    int numKeys = collationKeys.size();
+    List<Expression> orderByExpressions = new ArrayList<>(numKeys);
+    for (int i = 0; i < numKeys; i++) {
+      orderByExpressions.add(
+          convertOrderBy(collationKeys.get(i), collationDirections.get(i), collationNullDirections.get(i), pinotQuery));
     }
-    return orderByExpr;
+    return orderByExpressions;
   }
 
   private static Expression convertOrderBy(RexExpression rexNode, RelFieldCollation.Direction direction,
-      PinotQuery pinotQuery) {
-    Expression expression;
-    switch (direction) {
-      case DESCENDING:
-        expression = getFunctionExpression("DESC");
-        expression.getFunctionCall().addToOperands(toExpression(rexNode, pinotQuery));
-        break;
-      case ASCENDING:
-      default:
-        expression = getFunctionExpression("ASC");
-        expression.getFunctionCall().addToOperands(toExpression(rexNode, pinotQuery));
-        break;
+      RelFieldCollation.NullDirection nullDirection, PinotQuery pinotQuery) {
+    if (direction == RelFieldCollation.Direction.ASCENDING) {
+      Expression expression = getFunctionExpression("asc");
+      expression.getFunctionCall().addToOperands(toExpression(rexNode, pinotQuery));
+      if (nullDirection == RelFieldCollation.NullDirection.FIRST) {
+        Expression nullFirstExpression = getFunctionExpression("nullsfirst");
+        nullFirstExpression.getFunctionCall().addToOperands(expression);
+        return nullFirstExpression;
+      } else {
+        return expression;
+      }
+    } else {
+      Expression expression = getFunctionExpression("desc");
+      expression.getFunctionCall().addToOperands(toExpression(rexNode, pinotQuery));
+      if (nullDirection == RelFieldCollation.NullDirection.LAST) {
+        Expression nullFirstExpression = getFunctionExpression("nullslast");
```

Comment:
`nullLastExpression`?
also it means nullFirst/nullLast is added upon request only if the default behavior is different from Postgres --> if so we should add comment to clarify that.

Reply:
Good catch.
Added some notes also